ibm/resource_ibm_cse_instance.go

```diff
@@ -0,0 +1,388 @@
+				Type:        schema.TypeString,
+				Description: "The exposed tcp port range for the CSE.",
+				Optional:    true,
+			},
+			"udp_range": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The exposed udp port range for the CSE.",
+				Optional:    true,
+			},
+			"max_speed": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The maxium network speed for the CSE.",
+				Optional:    true,
+			},
+			"estado_proto": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The estado protocol for the CSE, value could be tcp, http or https.",
+				Optional:    true,
+			},
+			"estado_port": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The estado port for the CSE",
+				Optional:    true,
+			},
+			"estado_path": &schema.Schema{
+				Type:        schema.TypeString,
+				Description: "The estado path for the CSE",
+				Optional:    true,
+			},
+			"dedicated": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The value is 1 will create the CSE in dedicated device, is 0 will create in shared device.",
+				Optional:    true,
+			},
+			"multi_tenant": &schema.Schema{
+				Type:        schema.TypeInt,
+				Description: "The dedicated device will only run this CSE when the value is 1.",
+				Optional:    true,
+			},
+			"acl": &schema.Schema{
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "The acl setting for this CSE.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+				Set:         schema.HashString,
+			},
+			"url": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"service_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"static_addresses": &schema.Schema{
+				Type:     schema.TypeSet,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCSEInstanceCreate(d *schema.ResourceData, meta interface{}) error {
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	payload, err := genCreatePayload(d)
+	if err != nil {
+		return err
+	}
+
+	log.Printf("resourceCSEInstanceCreate: payload=%v\n", payload)
+
+	srvId, err := seAPI.CreateServiceEndpoint(payload)
+
+	if err != nil {
+		return err
+	}
+
+	d.SetId(srvId)
+	d.Set("service_id", srvId)
+
+	return nil
+}
+
+func resourceCSEInstanceRead(d *schema.ResourceData, meta interface{}) error {
+	cseClient, err := meta.(ClientSession).CseAPI()
+	if err != nil {
+		return err
+	}
+
+	seAPI := cseClient.ServiceEndpoints()
+
+	srvId := d.Get("service_id").(string)
+
+	srvObj, err := seAPI.GetServiceEndpoint(srvId)
+
+	if err != nil {
+		return err
+	}
+
+	d.Set("url", srvObj.Service.URL)
+	addresses := []string{}
+	for _, se := range srvObj.Endpoints {
+		addresses = append(addresses, se.StaticAddress)
+	}
+
+	d.Set("static_addresses", addresses)
```

Comment:
In Read please set back and all attributes(required,optional...) defined in the schema...
d.Set("service", srvObj.....)
d.Set("customer", srvObj...)
-------so on

This help when user tries to import the resource in to terraform . If the resource is created outside terraform via API or cli user can use terraform import command  to load the state .

Reply:
We already have the attributes be filled. Here is a test example:
```
{
    "version": 3,
    "terraform_version": "0.11.13",
    "serial": 3,
    "lineage": "dce79919-fd73-99d4-9e8e-69881ef08aa3",
    "modules": [
        {
            "path": [
                "root"
            ],
            "outputs": {},
            "resources": {
                "ibm_cse_instance.instance2": {
                    "type": "ibm_cse_instance",
                    "depends_on": [],
                    "primary": {
                        "id": "us-south-terraform-2-customer-2-shared-mt-1g-1bb1667f",
                        "attributes": {
                            "acl.#": "1",
                            "acl.2474229694": "10.10.10.1/24",
                            "customer": "customer-2",
                            "data_centers.#": "1",
                            "data_centers.2286138284": "dal10",
                            "dedicated": "0",
                            "estado_path": "/healthcheck",
                            "estado_port": "80",
                            "estado_proto": "http",
                            "id": "us-south-terraform-2-customer-2-shared-mt-1g-1bb1667f",
                            "max_speed": "1g",
                            "multi_tenant": "1",
                            "region": "us-south",
                            "service": "terraform-2",
                            "service_addresses.#": "1",
                            "service_addresses.1085037085": "10.102.33.131",
                            "service_id": "us-south-terraform-2-customer-2-shared-mt-1g-1bb1667f",
                            "static_addresses.#": "0",
                            "tcp_ports.#": "2",
                            "tcp_ports.80": "80",
                            "tcp_ports.8080": "8080",
                            "tcp_range": "8000-9000",
                            "udp_ports.#": "1",
                            "udp_ports.12345": "12345",
                            "udp_range": "50000-51000"
                        },
                        "meta": {},
                        "tainted": false
                    },
                    "deposed": [],
                    "provider": "provider.ibm"
                }
            },
            "depends_on": []
        }
    ]
}
```